elasticpath/react-pwa-reference-storefront
# GDPR enhancements

Description:
This PR captures some bug fixes, refactoring, and increased functionality surrounding managing GDPR and data policy consent.  These enhancements include:
- The `X-Ep-Data-Policy-Segments` header value is now exposed as a configuration instead of being hard-coded in the component
- The GDPR consent form now waits for all calls to complete before redirecting.
- The GDPR consent form now uses a zoom instead of making a number of GET requests equal to the number of data policies
- There is a new component which lives in the Profile Page for now.  It shows a list of data policies available based on store and `X-EP-Data-Policy-Segments`.  It also allows users to consent or revoke consent to any individual policy at any time.

I also removed the duplicated all-fields-required value in the fr-FR.json locale file.

Linting:

Component Updates:

Tests:
You will need to set up an EP Store with Data Policies enabled.  More information on configuring Data Policies in EP can be found here: https://documentation.elasticpath.com/commerce-manager/docs/configuration/commerce-configuration.html#data-policies
1. Enable GDPR in ep.config.json
2. Open the PWA in a private browser
3. You will be prompted by the GDPR consent form
4. Select the checkbox and click the ACCEPT button
5. All calls to Cortex will now complete before the page refreshes.  I have left the page refresh in, although I am not sure if it is necessary or if the modal just need to close.
6. Authenticate as a REGISTERED user, or create a new registered user.
7. Navigate to `/profile`
8. Scroll to the bottom of the page where the Data Polices section is displayed
9. Give and Revoke consent to data policies.

Documentation:
GDPR.dataPolicySegments will need to be added to https://documentation.elasticpath.com/storefront-react/docs/setup.html#configuration-parameter-descriptions.

More information on data policy segments and the `X-Ep-Data-Policy-Segments` header can be found here: https://documentation.elasticpath.com/commerce/docs/cortex/frontend-dev/managing-data-policy.html
